proto/backup.proto

```diff
@@ -185,8 +187,27 @@ message GCS {
     // You can get one from https://console.cloud.google.com/apis/credentials, and
     // copy the content, set it as string here.
     string credentials_blob = 6;
+    Bucket bucket_info = 7;
 }
 
+// S3 and GCS can be transitioned to use this
+message Bucket {
+    string endpoint = 1; // use a different endpoint (host + url)
+    string host = 2; // use a different host
+    string region = 3;
+    string bucket = 4;
+    string prefix = 5;
+    string storage_class = 6;
+}
+
+// CloudDynamic allows adding new fields without changing protobuf definitions
+message CloudDynamic {
+    Bucket bucket = 1;
+    string name = 2;
```

Comment:
```suggestion
    string service = 2;
```

`name` sounds like a file name here 🙃

Reply:
how about `provider_name` or `cloud_provider` or `cloud_name` ?

Reply:
also good

Reply:
okay, I changed it to "provider_name"